presenters/thread.rb

```diff
@@ -23,7 +23,7 @@ def initialize(thread, user, is_read, unread_count, is_endorsed)
     @is_endorsed = is_endorsed
   end
 
-  def to_hash with_responses=false, resp_skip=0, resp_limit=nil
+  def to_hash with_responses=false, resp_skip=0, resp_limit=nil, recursive=false
```

Comment:
Oh my! I have not. Thank you for pointing this out @jcdyer; this is huge.
We can either update this change and make default behaviour as recursive or we need to change all clients. 
I personally like the idea of keeping it non-recursive by default (to avoid huge payload when not required) but then we need to change its consumers.
I had updated 'discussion_api' for mobile according to this change but I am not sure what other clients do we have. Is there any document that we keep to list for such information?